File: sprint8/JS/lesson4code2.js
function find_all(text, pattern) {
    let occurrences = [];
    let start = 0; // Начнём поиск с начала строки.
    // Найдём первое вхождение, если оно есть.
    while ((pos = find(text, pattern, start)) !== -1) {
        occurrences.push(pos); // Сохраним вхождение в список.
        start = pos + 1;       // И продолжим поиск, начиная с позиции,
                               // следующей за только что найденной.
    }
    return occurrences;
}

File: sprint8/By lessons/lesson4code2/lesson4code2.swift
func findAll(text: String, pattern: String) -> [Int] {
    var occurrences = [Int]()
    var start = 0 // Начнём поиск с начала строки.
    // Найдём первое вхождение, если оно есть.
    while true {
        let pos = find(text: text, pattern: pattern, start: start)
        if pos == -1 {
            break
        }
        occurrences.append(pos) // Сохраним вхождение в список.
        start = pos + 1       // И продолжим поиск, начиная с позиции,
                              // следующей за только что найденной.
    }
    return occurrences
}
